.github/workflows/prune.yml

```diff
@@ -10,19 +10,74 @@ on:
     - cron: "0 0 * * *"  # every day at midnight
 
 jobs:
-  build:
+  # Phase 1: find all docker image names and release tags
+  fetch_image_info:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Fetch all aiSSEMBLE docker image names and release versions, then save to respective outputs
+        id: fetch-image-info
+        env:
+          GH_TOKEN: ${{ secrets.GHCR_IO_TOKEN }}
 
+        # Find all docker image names that starts with "aissemble" and does not end with "-chart" and save them to matrix-values output
+        # And find all release version tags and saves them to release-values output
+        run: |
+          image_names=$(gh api \
+            -H "Accept: application/vnd.github+json" \
+            -H "-GitHub-Api-Version: 2022-11-28" \
+            -H "Authorization: Token ${GH_TOKEN}" \
+            --paginate "/orgs/boozallen/packages?package_type=container" | \
+            jq -c '[.[] | select((.name | startswith("aissemble")) and (.name | endswith("-chart") | not)) | .name]')
+          
+          # save to output
+          echo "matrix_values=${image_names}" >> $GITHUB_OUTPUT
+  
+          # loop through each image name and find all release tags using regex pattern
+          for name in $(echo "$image_names" | jq -r '.[]'); do
+              release_versions=$(gh api \
+                -H "Accept: application/vnd.github+json" \
+                -H "-GitHub-Api-Version: 2022-11-28" \
+                -H "Authorization: Token ${GH_TOKEN}" \
+                --paginate "/orgs/boozallen/packages/container/${name}/versions" \
+                | jq -r '.[] | .metadata.container.tags[] | select(test("^\\d+\\.\\d+\\.\\d+$"))')
+          
+              # append all release tags to all_release_versions array
+              all_release_versions+=("${release_versions}")
+          done
+          
+          # replace all whitespaces with newline (required for sort) and then sort all release versions while removing any duplicates
+          all_release_versions_sorted=$(echo "${all_release_versions[@]}" | tr ' ' '\n'| sort -u )
+          
+          # replace the newlines with commas and then remove the trailing comma
+          all_release_versions=$(echo "${all_release_versions_sorted[@]}" | tr '\n' ',' | sed 's/,$//')
+          
+          # save to output
+          echo "release_values=${all_release_versions}" >> $GITHUB_OUTPUT
+          
+
+    # Identify the outputs of phase 1: fetch_image_info so they can be used in phase 2
+    outputs:
+      matrix-values: ${{ steps.fetch-image-info.outputs.matrix_values }}
+      release-values: ${{ steps.fetch-image-info.outputs.release_values }}
+
+  
+  # Phase 2: loop through output of phase 1 to prune images
+  prune_ghcr:
+    # Tells workflow to wait for the "fetch_image_info" phase to finish
+    needs: fetch_image_info
     runs-on: ubuntu-latest
 
+    strategy:
+      matrix:
+        package-name: ${{ fromJson(needs.fetch_image_info.outputs.matrix-values) }}
```

Comment:
I: Here, we're identifying the matrix values that will be used repeat the prune step (aka the list of docker image names). We have to repeat this step for each docker image because the `ghcr-cleanup-action` action only allows us to identify one package to prune at a time.

The matrix values are given as a JSON array containing the image names. For example,
```
["aissemble-vault","aissemble-quarkus","aissemble-airflow","aissemble-fastapi","aissemble-hive-service","aissemble-hive-mysql",...]
```